woocommerce/woocommerce-ios
# [Free Trial - Local notification] Stop scheduling expiry notifications

Part of: #10092


## Description
Stops scheduling the local notifications one day before and after the Free Trial expiry. 

**Notes**
- We are keeping the notification cancellation and handling the notification tapping code to ensure that previously scheduled notifications are handled/cancelled correctly. 
    - I didn't remove the previously scheduled notifications on app launch (for a free trial site) because we will not schedule the "24h after subscribed" notification for those cases. And it is better to at least keep those "one day before and after" notifications to increase purchases. 
- We can remove all the relevant code 30 days after this PR is released into production.  This is to ensure that all of the previously scheduled notifications are shown to the user and we correctly handle the tap on those notifications.

## Testing instructions

Prerequisites:
- Ensure you have access to a site running a trial plan that is not expired yet.
- Do a fresh installation of the app. This is to ensure that previously scheduled local notifications don't interfere testing. 

---

- Launch the app, log in, and switch to your test site if needed. For easier testing, please run the app on a physical device.
- Kill the app, open Settings, and change the time of the device to 1 day and 1 minute prior to the expiration date. The expiry date is at 0:00, so my expiry date is on May 19, then I should change the date to May 17, 23:59.
- Wait for 1 minute, and validate that you don't see any local notification.
- Kill the app, change the time of your device to 1 day and 59 seconds after the expiration date. E.g: May 19, 23:59, given the above example.
- Wait for 1 minute, and validate that you don't see any local notification.

Copied and altered the testing steps from - https://github.com/woocommerce/woocommerce-ios/pull/9719 Thanks, Huong. 😄 


## Screenshots
NA

---
